keras_nlp/src/tokenizers/byte_pair_tokenizer.py

```diff
@@ -64,12 +64,12 @@ def create_alts_for_unsplittable_tokens(unsplittable_tokens):
     # Create alternates for all special tokens that will be not split during
     # tokenization.
     alts = []
-    prefix = "Ĵ"
-    # Trim out splitters.
-    replace_pattern = r"'|\s+|[^\p{L}\p{N}]+"
-    for token in unsplittable_tokens:
-        token = re.sub(replace_pattern, "", token)
-        alts.append(prefix + token)
+    for index in range(len(unsplittable_tokens)):
+        # Map tokens to ĴA, ĴB, ĴC, etc. We assume these are usually unique.
+        prefix = "Ĵ"
+        digits = [int(d) for d in str(index)]
+        suffix = "".join([chr(ord("A") + d) for d in digits])
```

Comment:
This will make all the letters upper case, right? but the code it replaces doesn't do that. I was wondering if there is any specific reason for this change.

Reply:
> This will make all the letters upper case, right?

This just turn the digit of the special token to a very basic string of characters. So `0 -> A`, `1 -> B`, `11 -> BB`.

This was actually a bug I uncovered during this change. We are trying to escape special tokens here, so they don't get split into multiple tokens.

Before we were just stripping all punctuation and split-able characters from the speical token and prefix "Ĵ" which was a random uncommon character we had chosen. But this would break! It would map `<s>` and `</s>` to the exact same token id. Instead I am trying to make sure we map each unsplittable token to a unique character sequence. We can use literal digits, because this tokenizer split on digits. So I mapped each unsplittable token to a unique digit, and map each digit to a string of uppercase letters to avoid it splitting.

Fixes the bug, and this whole thing is gross, but I don't think this solution is any more or less gross that what was here before.

Reply:
Oh, I see! Thanks for the clarification and thanks for fixing the bug!

Reply:
added a few more comments to this effect